### ISSUE TITLE
Hunspell dictionaries not detected on Fedora 35 because they're installed to a different location

### ISSUE THREAD

**Issue by warrickball** [JohannesBuchner/languagecheck#1]
I'm running Fedora 35 and have installed the Python bindings for Hunspell as well as Hunspell and its dictionaries. When I run languagecheck on a file, I get the warning
```
analysis: spelling mistakes
   could not load dictionaries ( /usr/share/hunspell/en_GB.dic /usr/share/hunspell/en_GB.aff Is hunspell installed?
```
Indeed, that directory doesn't exist. Instead, though I don't know why, the files are installed to `/usr/share/myspell`:
```
$ find /usr/share/ -name en_GB.aff
/usr/share/myspell/en_GB.aff

$ dnf provides */en_GB.aff
Last metadata expiration check: 10 days, 15:12:50 ago on Sat 02 Apr 2022 19:33:47 BST.
R-hunspell-3.0.1-4.fc35~bootstrap.x86_64 : High-Performance Stemmer, Tokenizer, and Spell Checker
Repo        : fedora
Matched from:
Filename    : /usr/lib64/R/library/hunspell/dict/en_GB.aff

hunspell-en-GB-0.20140811.1-20.fc35.noarch : UK English hunspell dictionaries
Repo        : @System
Matched from:
Filename    : /usr/share/myspell/en_GB.aff

hunspell-en-GB-0.20140811.1-20.fc35.noarch : UK English hunspell dictionaries
Repo        : fedora
Matched from:
Filename    : /usr/share/myspell/en_GB.aff

texmaker-1:5.1.2-1.fc35.x86_64 : LaTeX editor
Repo        : fedora
Matched from:
Filename    : /usr/share/texmaker/en_GB.aff
```
I've succesffuly worked around this by changing the relevant paths in `language_check.py` but thought I'd flag this. I don't know how `languagecheck.py` might automatically detect where the dictionaries are installed.

**Comment by JohannesBuchner** [comment]
Ah, I guess we could try both folders (hunspell and myspell) before giving up.

**Comment by JohannesBuchner** [comment]
Can you try whether the latest commit solves this for you?

**Comment by warrickball** [comment]
It does, thanks!

FWIW, I was considering opening a PR with something like this, in case it turns out other systems also install the dictionaries to strange locations:
```Python
    for dir in ['hunspell', 'myspell']: # Fedora installs dictionary to /usr/share/myspell
        try:
            h = hunspell.HunSpell('/usr/share/%s/%s.dic' % (dir, hlang), '/usr/share/%s/%s.aff' % (dir, hlang)) # if this fails, you dont have that language installed or are not on Unix
            break
        except Exception:
            continue
    else:
        print("   could not load dictionaries (", '/usr/share/hunspell/%s.dic' % hlang, '/usr/share/hunspell/%s.aff' % hlang, "Is hunspell installed?")
        return
```

**Comment by JohannesBuchner** [comment]
for-else ... also nice. :)
